blockchain/validate.go

```diff
@@ -821,6 +841,44 @@ func CheckBlockSanity(block *dcrutil.Block, timeSource MedianTimeSource, chainPa
 	return checkBlockSanity(block, timeSource, BFNone, chainParams)
 }
 
+// isDCP0005Violation returns whether or not the block is known to violate
+// DCP0005.  Due to a bug that has since been fixed, some blocks were accepted
+// that violated DCP0005 by still being submitted on an old block version
+// (version 6 on mainnet) when the majority of the network had already upgraded
+// to the new version specified by DCP0005 (version 7 on mainnet).
+//
+// The blocks that violated DCP0005 due to specifying the old version were
+// otherwise valid and are whitelisted by this function so that they will be
+// accepted when fully syncing the chain without checkpoints.
+func isDCP0005Violation(network wire.CurrencyNet, header *wire.BlockHeader, blockHash *chainhash.Hash) bool {
+	// 430191 is the height of the last block on mainnet that violated DCP0005, so
+	// return false immediately if the height is greater than that.
+	if header.Height > 430191 {
```

Comment:
Granted this works because it also happens to be higher than the value for testnet, but I think this should probably be explicit for the network.  The same is true for the hash checks too even though a collision is nigh impossible.

What do you think about this?

e.g.
```Go
func isDCP0005Violation(network wire.CurrencyNet, header *wire.BlockHeader, blockHash *chainhash.Hash) bool {
	switch network {
	case wire.MainNet:
		// 430191 is the height of the last block on mainnet that violated
		// DCP0005, so return false immediately if the height is greater than
		// that.
		if header.Height > 430191 {
			return false
		}

		// Return whether or not the block is any of the following 5 mainnet
		// blocks that are known to violate DCP0005.
		return header.Height == 413762 && *blockHash == *block413762Hash ||
			header.Height == 414036 && *blockHash == *block414036Hash ||
			header.Height == 424011 && *blockHash == *block424011Hash ||
			header.Height == 428809 && *blockHash == *block428809Hash ||
			header.Height == 430191 && *blockHash == *block430191Hash

	case wire.TestNet3:
		// 323282 is the height of the last block on testnet that violated
		// DCP0005.  Return true if the height is less than or equal to 323282
		// so that the DCP0005 version check is not enforced until after that
		// point.
		return header.Height <= 323282
	}

	return false
}
```

Reply:
Comment from @rstaudt2:

> I can update it to that. I actually had something similar at first and then removed it to avoid the extra network check, but in practice, it shouldn't really have any performance impact, and this reads better.

Reply:
Updated.